src/cli/command_processor.py

```diff
@@ -142,22 +140,26 @@ def experiments_validate(self, path: Path) -> Tuple[bool, str]:
 
     @log_function
     def experiments_results(
-        self, all_results: bool, show: bool, path:Path
-    ) -> Optional[List[ResultType]]:
-        results = None
-        is_local = self.__local.is_experiment_local(path)
-
-        if is_local:
-            results = self.__local.get_results(path, all_results)
-        else:
-            results = self.__remote.get_results('path', all_results, block=True, timeout=100)
+        self, all_results: bool, path:Path
+    ) -> GeneratedResultType:
+        results: GeneratedResultType = self.__get_results(path=path)
+        return results
 
-        if show:
-            return results
+    @log_function
+    def __store_results(self, results: GeneratedResultType, path: Path) -> None:
+        print(f"store_results called with {results}")
+        processed_results_directory = path / "results"
+        if not processed_results_directory.is_dir():
+            processed_results_directory.mkdir(parents=True)
 
-        self.__store_results(results)
-        return None
+        processed_result_json_file = processed_results_directory / 'processed.json'
+        utils.write_json_file(processed_result_json_file, results, encoder_cls=utils.ComplexEncoder)
 
     @log_function
-    def __store_results(self, results: List[ResultType]) -> None:
-        pass
+    def __get_results(self, path: Path) -> GeneratedResultType:
+        processed_results_directory = path / "results"
+        if not processed_results_directory.is_dir():
+            raise ResultDirectoryNotAvailable(str(processed_results_directory))
+
+        processed_result_json_file = processed_results_directory / 'processed.json'
+        return cast(GeneratedResultType, utils.read_json_file(processed_result_json_file))
```

Comment:
I tested this scenario. When user tries to get the results of an experiment, then the user sees the exception (not found, or invalid json) instead of seeing the results. I think it should be fine.

Reply:
agree